Issue/7508 google signup progress

theck13 opened the pull request:
### Fix
Add a progress dialog in the background of the Google signup interface to indicate the process is still ongoing to solve the issue described in https://github.com/wordpress-mobile/WordPress-Android/issues/7508.

### Test
0. Clear app data.
1. Tap ***Sign Up for WordPress.com*** button.
2. Tap ***Sign Up with Google*** button.
3. Notice ***Signing up with Google...*** progress dialog is shown.
4. Tap back arrow.
5. Notice ***Signing up with Google...*** progress dialog is dismissed.
6. Tap ***Sign Up with Google*** button.
7. Notice ***Signing up with Google...*** progress dialog is shown.
8. Tap Google account from chooser dialog.
9. Notice ***Signing up with Google...*** progress dialog is still shown.

hypest commented:
I think I ran into a couple of cases where the popup misbehaves:

1. If you tap on "Sign up with Google" and rotate the device, the popup looks as if it's _behind_ the bottom action sheet. Can we perhaps just dismiss the actionsheet after we tap on one of its options?

2. If you dismiss the Google account selection dialog (without selecting an account) the popup is gone but it you rotate the device it comes back and can't be cancelled at all. The user is stuck there since the "Back" button is inhibited.

theck13 commented:
> 1. If you tap on "Sign up with Google" and rotate the device, the popup looks as if it's behind the bottom action sheet. Can we perhaps just dismiss the actionsheet after we tap on one of its options?

I'm unable to reproduce this, but the proposed solution goes against the design a little.  The bottom sheet is intentionally left shown after tapping the ***Sign Up with Google*** button for the case when the user cancels the Google account selection dialog on purpose.  That way they won't have to tap the ***Sign Up for WordPress.com*** button again to show the bottom sheet and choose ***Sign Up with Email*** or ***Sign Up with Google***.  They can just immediately choose ***Sign Up with Email*** or ***Sign Up with Google*** again.

> 2. If you dismiss the Google account selection dialog (without selecting an account) the popup is gone but it you rotate the device it comes back and can't be cancelled at all. The user is stuck there since the "Back" button is inhibited.

The `SignupGoogleFragment` was not being removed from the `FragmentManager` after dismissing the Google account selection dialog, e.g. tapping the back button.  I added the removal of the `SignupGoogleFragment` instance in aab46c0.

I also added the dismissal of the bottom sheet in `LoginActivity.onDestroy` (3d09c1f) and the progress dialog in `SignupGoogleFragment.onDetach` (39d4513), because they were causing leaked windows as show in the log snippet below.

```
android.view.WindowLeaked: Activity org.wordpress.android.ui.accounts.LoginActivity has leaked window DecorView@41f34fc[] that was originally added here
```

hypest commented:
Thanks for the changes @theck13 , the new commits have resolves the issues brought up so, this is 🚢it for me!
